runtime/extern.c

```diff
@@ -830,7 +832,7 @@ static void extern_rec(struct caml_extern_state* s, value v)
     }
     case Double_array_tag: {
       mlsize_t nfloats;
-      CAMLassert(sizeof(double) == 8);
+      static_assert(sizeof(double) == 8, "");
```

Comment:
Since you're turning these into `static_assert`, could you simply keep one and move it to the beginning of the function?

Reply:
Sure.